examples/UpdateContractMint/Program.cs

```diff
@@ -0,0 +1,260 @@
+        // information in the smart contract module schema.
+        var parameterBytes = Parameter.UpdateJson(
+            moduleSchema,
+            receiveName.GetContractName(),
+            receiveName.GetEntrypoint(),
+            jsonParameter
+        );
+
+
+        // Prepare the transaction for signing.
+        var updatePayload = new Concordium.Sdk.Transactions.UpdateContract(
+            amount,
+            contract,
+            receiveName,
+            parameterBytes
+        );
+        var sender = account.AccountAddress;
+        var sequenceNumber = client.GetNextAccountSequenceNumber(sender).Item1;
+        var expiry = Expiry.AtMinutesFromNow(30);
+        var maxEnergy = new EnergyAmount(options.MaxEnergy);
+        var preparedPayload = updatePayload.Prepare(sender, sequenceNumber, expiry, maxEnergy);
+
+        // Sign the transaction using the account keys.
+        var signedTransaction = preparedPayload.Sign(account);
+
+        // Submit the transaction.
+        var txHash = await client.SendAccountTransactionAsync(signedTransaction);
+        Console.WriteLine($"Successfully submitted transfer transaction with hash {txHash}");
+
+        // Watch the status of the transaction, until it becomes finalized in a block.
+        Console.WriteLine($"Waiting for the transaction to finalize...");
+        var finalized = await client.WaitUntilFinalized(txHash);
+
+        Console.WriteLine($"Finalized in block with hash {finalized.BlockHash}");
+
+        // Check whether the transaction was rejected and if so exit.
+        if (finalized.Summary.TryGetRejectedAccountTransaction(out var reason))
+        {
+            Console.WriteLine($"Transaction got rejected:\n{reason}");
+            return;
+        }
+        Console.WriteLine($"Transaction got accepted!");
+
+        // Exact the logs of updated smart contract instances from the outcome.
+        if (!finalized.Summary.TryGetContractUpdateLogs(out var updates))
+        {
+            throw new InvalidOperationException(
+                "Transaction summary failed to parse as a contract update transaction."
+            );
+        }
+        // Print out the events from each updated contract.
+        foreach (var update in updates)
+        {
+            var updatedContract = update.Item1;
+            Console.WriteLine($"Contract {updatedContract} logged:");
+            // If this is our contract, then we use the module schema to parse the events.
+            if (updatedContract == contract)
+            {
+                foreach (var evt in update.Item2)
+                {
+                    // Use the smart contract module schema to deserialize the events.
+                    var json = evt.GetDeserializeEvent(moduleSchema, receiveName.GetContractName());
+                    Console.WriteLine($"- {json}");
+                }
+            }
+            else
```

Comment:
Are other contracts relevant?

Reply:
No not really. Initially I just assumed our contract is the only one, but then I had to provide a complex comment explaining this and what to do when it is not the case, so I choose instead to make the example more general.